### PR TITLE
Incineratez/role permissions fix

### PR DESCRIPTION
**Pull request by IncinerateZ**
fixes issue with [p]!role create _name_ _[options]_ where an error is thrown when trying to add a Permissions option.

fixes issue with [p]!role temporary create _name_ _ticks_ _[options]_ where an error is thrown when trying to run the command.

major changes: 
- option "Permissions" renamed to "Perms".
- usage of Perms option modified to: `... "rolename" Perms: "permission1 permission2 permission3 ...."`

### CODE REVIEW

**Comment by msbukkuri** [comment]
@IncinerateZ Thanks for putting this together! I would recommend splitting off your fix for the "mentions everyone" permission into a different pull request. Because it is out of scope for this issue. 

I can create a separate issue for it, I'd stash/save that specific commit and assign it to that. Does that make sense?

File: Zhongli.Bot/Modules/RoleModule.cs [hunk 1]
```diff
@@ -241,6 +248,27 @@ public Task ToggleRoleAsync(IGuildUser user, IRole role)
...
+        
+        private GuildPermissions CreateGuildPermissions(string permissions) 
+        {
+            GuildPermissions gperms = new GuildPermissions(createInstantInvite: true, addReactions: true, stream: true, sendMessages: true, sendTTSMessages: true, embedLinks: true, attachFiles: true, readMessageHistory: true, mentionEveryone: true, useExternalEmojis: true, connect:true, speak: true, changeNickname: true);
+
+            if(!Equals(permissions, "")) 
+            {
+                string[] perms = permissions.Split(" ");
+
+                var dict = new Dictionary<String, Boolean>();
```
Comment (sabihoshi) [inline Zhongli.Bot/Modules/RoleModule.cs:260]: Even if this won't be needed, prefer to use the aliases of primitive types, i.e. `string` and `bool`.

File: Zhongli.Bot/Modules/RoleModule.cs [hunk 2]
```diff
@@ -241,6 +248,27 @@ public Task ToggleRoleAsync(IGuildUser user, IRole role)
...
...s: dict.GetValueOrDefault("managemessages", false), /*embedLinks: dict.GetValueOrDefault("embedlinks", false),*/ /*attachFiles: dict.GetValueOrDefault("attachfiles", false),*/ /*readMessageHistory: dict.GetValueOrDefault("readmessagehistory", false),*/ /*mentionEveryone: dict.GetValueOrDefault("mentioneveryone", false),*/ /*useExternalEmojis: dict.GetValueOrDefault("useexternalemojis", false),*/ /*connect: dict.GetValueOrDefault("connect", false),*/ /*speak: dict.GetValueOrDefault("speak", false),*/ muteMembers: dict.GetValueOrDefault("mutemembers", false), deafenMembers: dict.GetValueOrDefault("deafenmembers", false), moveMembers: dict.GetValueOrDefault("movemembers", false), useVoiceActivation: dict.GetValueOrDefault("usevoiceactivation", false), prioritySpeaker: dict.GetValueOrDefault("priorityspeaker", false), /*stream: dict.GetValueOrDefault("stream", false),*/ /*changeNickname: dict.GetValueOrDefault("changenickname", false),*/ manageNicknames: dict.GetValueOrDefault("managenicknames", false), manageRoles: dict.GetValueOrDefault("manageroles", false), manageWebhooks: dict.GetValueOrDefault("managewebhooks", false), manageEmojis: dict.GetValueOrDefault("manageemojis", false));
```
Comment (sabihoshi) [inline Zhongli.Bot/Modules/RoleModule.cs:267]: This is a bit messy

**Review by sabihoshi — CHANGES_REQUESTED** [review]

**Comment by sabihoshi** [comment]
Make sure your PR name is consistent with the spec of conventional commits, and there should be a space afterwards the `:` and proper capitalization.

The usage of `chore` is wrong in here since it's supposed to be `fix` as it fixes some functionality.

I need to add such guidelines on length, but prefer the length of commits to be at most 75 characters so they don't get cut off.

**Pull request merged**